# Once when engine was crashes after some memory corruptions in other place
# It was crashes somewhere here... some workaround checks...
# Probably it is wrong
--- mozilla/layout/style/nsStyleStruct.cpp.orig	2007-05-16 18:51:38.000000000 +0300
+++ mozilla/layout/style/nsStyleStruct.cpp	2007-06-19 14:43:41.000000000 +0300
@@ -1577,21 +1578,21 @@ nsStyleUserInterface::nsStyleUserInterfa
   mUserInput(aSource.mUserInput),
   mUserModify(aSource.mUserModify),
   mUserFocus(aSource.mUserFocus),
   mCursor(aSource.mCursor)
 { 
   CopyCursorArrayFrom(aSource);
 }
 
 nsStyleUserInterface::~nsStyleUserInterface(void) 
 { 
-  delete [] mCursorArray;
+  DELETE_ARRAY_IF(mCursorArray);
 }
 
 nsChangeHint nsStyleUserInterface::CalcDifference(const nsStyleUserInterface& aOther) const
 {
   nsChangeHint hint = nsChangeHint(0);
   if (mCursor != aOther.mCursor)
     NS_UpdateHint(hint, nsChangeHint_UpdateCursor);
 
   // We could do better. But it wouldn't be worth it, URL-specified cursors are
   // rare.
--- mozilla/layout/base/nsFrameManager.cpp.orig	2007-03-27 18:33:40.000000000 +0300
+++ mozilla/layout/base/nsFrameManager.cpp	2007-06-19 14:20:23.000000000 +0300
@@ -193,20 +193,21 @@ public:
       mStyle(aStyle),
       mNext(nsnull)
   {
     MOZ_COUNT_CTOR(UndisplayedNode);
   }
 
   NS_HIDDEN ~UndisplayedNode()
   {
     MOZ_COUNT_DTOR(UndisplayedNode);
     delete mNext;
+    mNext = nsnull;
   }
 
   nsCOMPtr<nsIContent>      mContent;
   nsRefPtr<nsStyleContext>  mStyle;
   UndisplayedNode*          mNext;
 };
 
 class nsFrameManagerBase::UndisplayedMap {
 public:
   UndisplayedMap(PRUint32 aNumBuckets = 16) NS_HIDDEN;
